addons/rmsmartshape/shapes/shape.gd

```diff
@@ -843,8 +843,14 @@ func generate_collision_points() -> PackedVector2Array:
 func bake_collision() -> void:
 	if not _collision_polygon_node:
 		return
-	var xform := _collision_polygon_node.get_global_transform().inverse() * get_global_transform()
-	_collision_polygon_node.polygon = xform * generate_collision_points()
+
+	var collision_points := generate_collision_points()
+
+	# Collision polygon is a child node -> don't apply transform
+	if str(_collision_polygon_node.get_path()).begins_with(str(get_path())):
```

Comment:
Wouldn't it be better to use `Node.is_ancestor_of()` here?

Reply:
Right as I was about to say it lol

Reply:
Yes, absolutely. I remembered there was something of the like, but couldn't find it.

Reply:
Did some benchmarking and my approach is actually faster than `is_ancestor_of`.
```gdscript
# 7.733ms
for i in range(1000):
    var is_child := str(node.get_path()).begins_with(str(get_path()))

# 15.917ms
for i in range(1000):
    var is_child := self.is_ancestor_of(node)
```

Reply:
Interesting, wouldn't guess at a glance.